src/bun.js/ws.exports.js

```diff
@@ -6,19 +6,51 @@ import EventEmitter from "events";
 import http from "http";
 
 const kBunInternals = Symbol.for("::bunternal::");
+const readyStates = ["CONNECTING", "OPEN", "CLOSING", "CLOSED"];
 
 class BunWebSocket extends globalThis.WebSocket {
   constructor(url, ...args) {
     super(url, ...args);
     this.#wrappedHandlers = new WeakMap();
   }
+  #binaryType;
   #wrappedHandlers = new WeakMap();
 
+  get binaryType() {
+    return this.#binaryType;
+  }
+  set binaryType(type) {
+    if (type !== "nodebuffer" && type !== "buffer" && type !== "blob" && type !== "arraybuffer") {
+      throw new TypeError("binaryType must be either 'blob', 'arraybuffer', 'nodebuffer' or 'buffer'");
+    }
+    this.#binaryType = type;
+  }
   on(event, callback) {
     if (event === "message") {
-      var handler = ({ data }) => {
+      var handler = ({ message }) => {
         try {
-          callback(data);
+          if (typeof message === "string") {
+            if (this.#binaryType === "arraybuffer") {
+              const encoder = new TextEncoder();
```

Comment:
can avoid this overhead by re-using the `TextEncoder` instance at the top

Reply:
actually

we can set `binaryType` on the WebSocket itself.

Reply:
right?

Reply:
We only support `arraybuffer` rn on the `WebSocket.cpp`